javascript/ql/src/experimental/Security/CWE-347-noVerification/JsonWebToken.ql

```diff
@@ -0,0 +1,65 @@
+/**
+ * @name JWT missing secret or public key verification
+ * @description The application does not verify the JWT payload with a cryptographic secret or public key.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 8.0
+ * @precision high
+ * @id js/jwt-missing-verification-jsonwebtoken
+ * @tags security
+ *       external/cwe/cwe-347
+ */
+
+import javascript
+import DataFlow::PathGraph
+
+DataFlow::Node unverifiedDecode() {
```

Comment:
You've copy-pasted some of this code throughout multiple files.  
You can move those predicates to a `.qll` file that you then import from multiple `.ql` files.